ServerCore/Areas/Identity/Pages/Account/Manage/Index.cshtml.cs

```diff
@@ -0,0 +1,176 @@
+        private readonly IEmailSender _emailSender;
+        private readonly PuzzleServerContext _context;
+
+        public IndexModel(
+            UserManager<IdentityUser> userManager,
+            SignInManager<IdentityUser> signInManager,
+            IEmailSender emailSender,
+            PuzzleServerContext context)
+        {
+            _userManager = userManager;
+            _signInManager = signInManager;
+            _emailSender = emailSender;
+            _context = context;
+        }
+
+        public string Username { get; set; }
+
+        public bool IsEmailConfirmed { get; set; }
+
+        [TempData]
+        public string StatusMessage { get; set; }
+
+        [BindProperty]
+        public InputModel Input { get; set; }
+
+        public class InputModel
+        {
+            [Required]
+            [EmailAddress]
+            public string Email { get; set; }
+
+            [Phone]
+            [Display(Name = "Phone number")]
+            public string PhoneNumber { get; set; }
+
+            [Display(Name = "Employee alias")]
+            public string EmployeeAlias { get; set; }
+            public string Name { get; set; }
+
+            [Display(Name = "T-shirt size")]
+            public string TShirtSize { get; set; }
+
+            [Display(Name = "User is visible to other users")]
+            public bool VisibleToOthers { get; set; }
+        }
+
+        public async Task<IActionResult> OnGetAsync()
+        {
+            var user = await _userManager.GetUserAsync(User);
+            if (user == null)
+            {
+                return NotFound($"Unable to load user with ID '{_userManager.GetUserId(User)}'.");
+            }
+
+            PuzzleUser puzzleUser = PuzzleUser.GetPuzzleUser(user.Id, _context);
+
+            var userName = await _userManager.GetUserNameAsync(user);
+
+            Username = userName;
+
+            Input = new InputModel
+            {
+                Email = puzzleUser.EmailAddress,
+                PhoneNumber = puzzleUser.PhoneNumber,
+                EmployeeAlias = puzzleUser.EmployeeAlias,
+                Name = puzzleUser.Name,
+                TShirtSize = puzzleUser.TShirtSize,
+                VisibleToOthers = puzzleUser.VisibleToOthers
+            };
+
+            IsEmailConfirmed = await _userManager.IsEmailConfirmedAsync(user);
+
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
+            var user = await _userManager.GetUserAsync(User);
+            if (user == null)
+            {
+                return NotFound($"Unable to load user with ID '{_userManager.GetUserId(User)}'.");
+            }
+
+            PuzzleUser puzzleUser = PuzzleUser.GetPuzzleUser(user.Id, _context);
+
+            if (Input.EmployeeAlias != puzzleUser.EmployeeAlias)
```

Comment:
Removing the InputModel would get rid of all of the field-by-field boilerplate here

Reply:
Good point. I'll see if it breaks everything else to use a PuzzleUser.